### ISSUE TITLE
1.1.1 Sidecars routing all 5432 traffic to single place regardless of hostname used.

### ISSUE THREAD

**Issue by pmcfadden** [istio/istio#13349]
**Describe the bug**
We have several RDS databases in AWS (running in the same VPC as our cluster with istio). We noticed after the upgrade that all traffic from inside the cluster to port 5432, regardless of host, ends up at only one of those databases. (even connecting to www.google.com on 5432 ends up at that same db).

**Expected behavior**
Expected to route to the database specified by the DNS name (and getting rejected from connecting to www.google.com)

**Steps to reproduce the bug**
Have 2 external postgres DBs. (have some way to tell them apart, such as table called db1 in first db, and table called db2 in the second) Create Services of type ExternalName that points to the DNS of each DB. 
Create postgres pod with sidecar
Exec into pod and try and connect to db1/db2/www.google.com
You should end up at only one of the two each time

**Version Istio**
client version: version.BuildInfo{Version:"1.1.1", GitRevision:"2b1331886076df103179e3da5dc9077fed59c989", User:"root", Host:"7077232d-4c6c-11e9-813c-0a580a2c0506", GolangVersion:"go1.10.4", DockerHub:"docker.io/istio", BuildStatus:"Clean", GitTag:"1.1.0-17-g2b13318"}
citadel version: version.BuildInfo{Version:"1.1.1", GitRevision:"2b1331886076df103179e3da5dc9077fed59c989-dirty", User:"root", Host:"7077232d-4c6c-11e9-813c-0a580a2c0506", GolangVersion:"go1.10.4", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.1.0-17-g2b13318"}
egressgateway version: version.BuildInfo{Version:"1.1.1", GitRevision:"2b1331886076df103179e3da5dc9077fed59c989", User:"root", Host:"7077232d-4c6c-11e9-813c-0a580a2c0506", GolangVersion:"go1.10.4", DockerHub:"docker.io/istio", BuildStatus:"Clean", GitTag:"1.1.0-17-g2b13318"}
galley version: version.BuildInfo{Version:"1.1.1", GitRevision:"2b1331886076df103179e3da5dc9077fed59c989-dirty", User:"root", Host:"7077232d-4c6c-11e9-813c-0a580a2c0506", GolangVersion:"go1.10.4", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.1.0-17-g2b13318"}
ilbgateway version: version.BuildInfo{Version:"", GitRevision:"", User:"", Host:"", GolangVersion:"", DockerHub:"", BuildStatus:"", GitTag:""}
ingressgateway version: version.BuildInfo{Version:"1.1.1", GitRevision:"2b1331886076df103179e3da5dc9077fed59c989", User:"root", Host:"7077232d-4c6c-11e9-813c-0a580a2c0506", GolangVersion:"go1.10.4", DockerHub:"docker.io/istio", BuildStatus:"Clean", GitTag:"1.1.0-17-g2b13318"}
pilot version: version.BuildInfo{Version:"1.1.1", GitRevision:"2b1331886076df103179e3da5dc9077fed59c989-dirty", User:"root", Host:"7077232d-4c6c-11e9-813c-0a580a2c0506", GolangVersion:"go1.10.4", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.1.0-17-g2b13318"}
policy version: version.BuildInfo{Version:"1.1.1", GitRevision:"2b1331886076df103179e3da5dc9077fed59c989-dirty", User:"root", Host:"7077232d-4c6c-11e9-813c-0a580a2c0506", GolangVersion:"go1.10.4", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.1.0-17-g2b13318"}
sidecar-injector version: version.BuildInfo{Version:"1.1.1", GitRevision:"2b1331886076df103179e3da5dc9077fed59c989-dirty", User:"root", Host:"7077232d-4c6c-11e9-813c-0a580a2c0506", GolangVersion:"go1.10.4", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.1.0-17-g2b13318"}
telemetry version: version.BuildInfo{Version:"1.1.1", GitRevision:"2b1331886076df103179e3da5dc9077fed59c989-dirty", User:"root", Host:"7077232d-4c6c-11e9-813c-0a580a2c0506", GolangVersion:"go1.10.4", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.1.0-17-g2b13318"}

**Version K8S**
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.3", GitCommit:"721bfa751924da8d1680787490c54b9179b1fed0", GitTreeState:"clean", BuildDate:"2019-02-04T04:49:22Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.8-eks-7c34c0", GitCommit:"7c34c0d2f2d0f11f397d55a46945193a0e22d8f3", GitTreeState:"clean", BuildDate:"2019-03-01T22:49:39Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}

**Installation**
helm template
"--set gateways.istio-egressgateway.enabled=true " \
            "--set grafana.enabled=true " \
            "--set tracing.enabled=true " \
            "--set kiali.enabled=true " \
            "--set global.disablePolicyChecks=false " \

**Environment**
EKS on AWS

**More Notes**
We realized we were missing ServiceEntries and added those, originally pointing them towards the cluster dns (postgres.namespace.svc.cluster.local). That didn't seem to get us very far. Then we tried pointing them towards the AWS given dns. that also did not work, until we deleted the "Kind: Service" objects and re-added them. Then things seem to route appropriately.

**Comment by howardjohn** [comment]
@philrud did some work with ExternalName - is this a known issue, Phil?

It would be useful if you could provide the config dump of one of you envoys - localhost:15000/config_dump. If you need to you can filter it to just the listener for port 5432.

**Comment by philrud** [comment]
Creating "ExternalName"-typed service is not enough/not required to add an external service to the mesh. I'd recommend removing "ExternalName" service if possible – you're looking in the right direction, you need a ServiceEntry (and potentially a VirtualService).

Here's the corresponding docs section on this: https://istio.io/docs/concepts/traffic-management/#service-entries

Hope this helps. If not – please also share yamls of your services and Istio k8s resources (ServiceEntries, VirtualServices, etc.)

**Comment by pmcfadden** [comment]
[config_dump.tar.gz](https://github.com/istio/istio/files/3085204/config_dump.tar.gz)


- I've attached 2 different config dumps as well as the service/serviceentry yamls. The outbound working file is the config where I've applied the serviceentries, then the services. connecting from the pgclient pod (where these configs came from) to both postgres and pg2 works as I had expected, ending up at 2 separate databases. 
- I then deleted the serviceentries and re-added them. the results of the config dump are in outbound-going-to-postgres. Now regardless of what hostname I use from the pgclient pod (pg2, postgres, www.google.com) I end up at the "postgres" service.
- FWIW if I then delete the services and re-add them now i end up at the right dbs aagain, and www.google.com fails.

@philrud thanks for the info. We can look at removing our externalname services, though we were hopeful we could keep them so that apps can connect to "postgres" as opposed to"name-that can change-over-time.aws.amazon.com" But probably not a deal-breaker if that's what it takes to keep everything happy.
